### PR TITLE
[20573] Squash version release notes

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description

This PR includes the following improvements:
- Squash release notes in minor versions
- Establish a more visible difference between the currently Fast DDS supported versions and the EOL versions
- Includes links of each documentation version (from v1.6.0 onwards)
- Fixes a typo introduced in v2.12.1 and dragged until v2.13.2

This PR has been built on top of (and needs to be merged after):
- #652 

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- **N/A** Code snippets related to the added documentation have been provided.
- [x] Documentation tests pass locally.
- **N/A** Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] CI passes without warnings or errors.

### CODE REVIEW

File: docs/notes/previous_versions/v2.9.0.rst [hunk 1]
```diff
@@ -18,7 +18,7 @@ fixes.
     It is also advisable to regenerate the type support from the IDL files using
     `Fast DDS-Gen v2.3.0 <https://github.com/eProsima/Fast-DDS-Gen/releases/tag/v2.3.0>`_.
     Furthermore, if upgrading to v2.9.0, it is also recommended to upgrade Fast CDR to
-    `v1.0.26 <https://github.com/eProsima/Fast-CDR/releases/tag/v1.0.26>`_.
+    `v1.0.06 <https://github.com/eProsima/Fast-CDR/releases/tag/v1.0.06>`_.
```
Comment (EduPonz) [inline docs/notes/previous_versions/v2.9.0.rst:21]: I don't think this is correct

File: docs/notes/notes.rst [hunk 2]
```diff
@@ -5,115 +5,25 @@
...
-5. Fix doxygen docs warnings. Prepare for compiling with Doxygen 1.10.0
-6. Add missing TypeLookup listeners
-7. Restore Blackbox tests names
-8. Add macOS Github CI
-9. Set 2.11.x as EOL
+    For further information about the corresponding versions of other products related to this Fast DDS version, please
+    refer to the :ref:`dependencies_compatibilities_product_compatibility` section.
 
-.. note::
+Supported versions
```
Comment (EduPonz) [inline docs/notes/notes.rst:15]: I think I'd actually put both supported versions and EOL versions at the same level as the latest in the TOC. I'd be something like:
1. Dependencies and compatibility
2. Version 2.13.3 (latest)
3. Supported versions
4. Previous end-of-life versions

File: docs/notes/previous_versions/v2.13.3.rst [hunk 3]
```diff
@@ -0,0 +1,25 @@
+`Version 2.13.3 <https://fast-dds.docs.eprosima.com/en/v2.13.3/index.html>`_
```
Comment (EduPonz) [inline docs/notes/previous_versions/v2.13.3.rst:1]:
```suggestion
`Version 2.13.3 (latest) <https://fast-dds.docs.eprosima.com/en/v2.13.3/index.html>`_
```

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Comment by JesusPoderoso** [comment]
@mergifyio rebase

File: docs/notes/previous_versions/v2.9.0.rst (hunk 1 above)
Comment (EduPonz) [reply docs/notes/previous_versions/v2.9.0.rst:21]: Yes, but the PR is changing it to `v1.0.06`, that's what I meant

Comment (JesusPoderoso) [reply docs/notes/previous_versions/v2.9.0.rst:21]: 😱

**Review by EduPonz — APPROVED** [review]

**Comment by EduPonz** [comment]
@richiprosima please test this

**Pull request merged**